features/step_definitions/check_provider_answers_steps.rb

```diff
@@ -9,3 +9,72 @@
 Then("the \"Partner details\" check your answers section should contain:") do |table|
   expect_questions_and_answers_in(selector: "#partner-details-questions", expected: table)
 end
+
+Given(/^I have created an application with (.*) proceedings (with|without) delegated functions$/) do |proceedings, df|
+  proceeding_array = proceedings.split(",").map(&:to_sym)
+  df_options = df == "with" ? proceeding_array.map { |p| { p.upcase => [10.days.ago, 10.days.ago] } }[0] : {}
+  applicant = create(
+    :applicant,
+    first_name: "Test",
+    last_name: "Test",
+    national_insurance_number: "JA123456A",
+    date_of_birth: "01-01-1970",
+    email: "test@test.com",
+    has_partner: false,
+  )
+  create(
+    :address,
+    address_line_one: "Transport For London",
+    address_line_two: "98 Petty France",
+    city: "London",
+    postcode: "SW1H 9EA",
+    lookup_used: true,
+    applicant:,
+  )
+  @legal_aid_application = create(
+    :legal_aid_application,
+    :with_non_passported_state_machine,
+    :at_entering_applicant_details,
+    :with_proceedings,
+    :with_delegated_functions_on_proceedings,
+    explicit_proceedings: proceeding_array,
+    set_lead_proceeding: proceeding_array.first,
+    df_options:,
+    applicant:,
+  )
+  login_as @legal_aid_application.provider
+end
+
+And("I view the check provider answers page") do
+  visit(providers_legal_aid_application_check_provider_answers_path(@legal_aid_application))
+  steps %(Then I should be on a page showing 'Check your answers')
+end
+
+Then("the {string} proceeding check your answers section should contain:") do |proceeding_code, table|
+  expect_questions_and_answers_in(selector: "##{proceeding_code.downcase}-questions", expected: table)
+end
+
+Then(/^the (.*) answer for (.*) (should|should not) match (.*)$/) do |question, proceeding_code, match, regex|
```

Comment:
if this is proceeding specific it might be clearer to have the word `proceeding` in the step
```suggestion
Then(/^the (.*) answer for proceeding (.*) (should|should not) match (.*)$/) do |question, proceeding_code, match, regex|
```

or if not proceeding specific then rename `proceeding_code` argument

Reply:
For now, it is only used for proceedings, it could maybe be extended in the future to other check answer blocks

Amended to ```ruby
the (.*) answer for the (.*) proceeding (should|should not) match
```
To allow the separation of the .* match with the should/should not